### PR TITLE
HDDS-9446. Set HadoopRPC as the default transport between S3 and Ozone.

### PR DESCRIPTION
**Pull request by duongkame**
## What changes were proposed in this pull request?

Set HadoopRPC as the default transport between S3 and Ozone. Details in JIRA.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9446

## How was this patch tested?

Tested locally to verify HadoopRPC is used for the traffic S3->OM.

### CODE REVIEW

**Comment by duongkame** [comment]
can you have a look at this @xBis7?

**Comment by kerneltime** [comment]
Ref: https://github.com/apache/ozone/pull/3820/files which switched default to GRPC. 

Can we just revert that change?

**Review by xBis7 — COMMENTED** [review]
@duongkame Thanks for the patch. The failing robot test in your CI [here](https://github.com/duongkame/ozone/actions/runs/6565936688/job/17836507083), it's checking for the config and if the transport is `GrpcOmTransport`, then it checks the metrics. 

Something causes the docker setup to override the transport config. How about reverting this change [here](https://github.com/apache/ozone/pull/3820/files#diff-c53af107c5adbe3d7f4b65b30ad7811fd42fe5f5eb4424f30018574beed0c53cL127)?

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 1]
```diff
@@ -2859,12 +2859,22 @@
   </property>
   <property>
     <name>ozone.om.transport.class</name>
-    <value>org.apache.hadoop.ozone.om.protocolPB.GrpcOmTransportFactory</value>
+    <value>org.apache.hadoop.ozone.om.protocolPB.Hadoop3OmTransportFactory</value>
     <tag>OM, MANAGEMENT</tag>
     <description>
-      Enable Ozone Manager S3G Grpc channel.
+      Property to determine the transport protocol for the client to Ozone Manager channel.
+      If GRPC (Hadoop3OmTransportFactory) is used, please make sure ozone.om.s3.grpc.server_enabled is enabled
```
Comment (ivandika3) [inline hadoop-hdds/common/src/main/resources/ozone-default.xml:2866]: Nit: `Hadoop3OmTransportFactory` supposed to be `GrpcOmTransportFactory`?

**Comment by duongkame** [comment]
@kerneltime @xBis7  The change in HDDS-7309 didn't only make the GRPC the default transport but also made it a viable option. Reverting it entirely would prevent others from intentionally setting GRPC as the transport protocol.

I just realized the grpc robot test relies on the fact that GRPC is enabled on OM side. So, I'm making the minimal change to just set HadoopRPC as the default transport from the client-side (S3). @xBis7

**Comment by xBis7** [comment]
@duongkame Thanks for the changes. I tested it locally and just changing the transport, is sufficient for switching to hRPC for S3G. 

I used the metrics to verify it, `GrpcMetrics` are getting registered, since OM S3 GRPC server is enabled but when producing traffic through S3G, there are no number changes. Instead, we can see `RpcActivityForPort9862` metrics getting updated.

With this patch do you see a performance change in your testing?

**Review by adoroszlai — APPROVED** [review]

**Pull request merged**